bespin/amazon/s3.py

```diff
@@ -43,147 +46,39 @@ def s3_location(self, value):
-            key.delete()
+        return bucket.objects.filter(Prefix=query.key[1:])
 
     def upload_file_to_s3(self, source_filename, destination_path):
-        source_size = os.stat(source_filename).st_size
-        if source_size > 30000000:
-            self.upload_file_to_s3_in_parts(source_filename, destination_path)
-        else:
-            self.upload_file_to_s3_as_single(source_filename, destination_path)
-
-    def upload_file_to_s3_in_parts(self, source_filename, destination_path):
-        source_file = open(source_filename, 'rb')
-        destination_file = self.s3_location(destination_path)
-
-        source = os.path.abspath(source_file.name)
-        source_size = os.stat(source).st_size
-        log.info("Uploading from %s (%s) to %s in parts", source, humanize.naturalsize(source_size), destination_file.full)
-
-        bucket = self.get_bucket(destination_file.bucket)
-
-        try:
-            with self.a_multipart_upload(bucket, destination_file.key) as mp:
-                for chunk, offset, length in self.determine_chunks(source_size, min_chunk=5242881):
-                    with FileChunkIO(source, 'r', offset=offset, bytes=length) as fp:
-                        log.info("Uploading chunk %s (%s)", chunk+1, humanize.naturalsize(length))
-                        mp.upload_part_from_file(fp, part_num=chunk+1)
-
-        except boto.exception.S3ResponseError as error:
-            if error.status is 403:
-                log.error("Seems you are unable to edit this location :(")
-                sys.exit(1)
-            else:
-                raise
-
-        log.info("Finished uploading")
-
-    def determine_chunks(self, total_size, min_chunk=5242881):
-        offset = 0
-        offsets = []
-        while offset < total_size:
-            if total_size - offset < min_chunk:
-                break
-            offsets.append(offset)
-            offset += min_chunk
-        offsets.append(total_size)
-        log.info("Broken up into %s chunks", len(offsets) - 1)
-
-        for index, offset in enumerate(offsets[:-1]):
-            nxt = offsets[index+1]
-            yield index, offset, nxt - offset
-
-    def upload_file_to_s3_as_single(self, source_filename, destination_path):
-        destination_file = self.s3_location(destination_path)
-
         source = os.path.abspath(source_filename)
         source_size = os.stat(source).st_size
-        log.info("Uploading from %s (%s) to %s in one go", source, humanize.naturalsize(source_size), destination_file.full)
+        dest = self.s3_location(destination_path)
 
-        bucket = self.get_bucket(destination_file.bucket)
-        key = Key(bucket)
-        key.name = destination_file.key
-        key.set_contents_from_filename(source_filename, policy='authenticated-read')
-        key.close()
+        log.info("Uploading from %s (%s) to %s", source, humanize.naturalsize(source_size), dest.full)
+        self.conn.upload_file(source, dest.bucket, dest.key[1:])
```

Comment:
I can't remember why I cared enough to upload "large" files in parts and I wonder if the need for that goes away with boto3 or not...

Also, I'm not sure why set policy='authenticated-read' in upload_file_to_s3_as_single

Reply:
You need to use multipart uploads for files larger than a certain amount. Botocore handles that via boto3.s3.transfer which upload_file() wraps.

Reply:
right, seems I chose 30mb arbitrarily though https://github.com/delfick/bespin/commit/a7c4ec7ae45f0779294a150014e3772c97501b98